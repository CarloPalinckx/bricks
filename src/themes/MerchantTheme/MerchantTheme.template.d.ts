import { StatelessComponent } from 'react';
import ThemeType from '../../types/ThemeType';
declare type PropsType = {
    theme: ThemeType;
};
declare const MerchantTheme: StatelessComponent<PropsType>;
export default MerchantTheme;
export { PropsType };
