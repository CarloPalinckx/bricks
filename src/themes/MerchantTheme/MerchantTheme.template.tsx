import React, { StatelessComponent } from 'react';
import ThemeType from '../../types/ThemeType';
import { ThemeProvider } from '../../utility/styled';

type PropsType = {
    theme:ThemeType;
};

const MerchantTheme:StatelessComponent<PropsType> = (props):JSX.Element => (
    <ThemeProvider theme={props.theme}>
        {props.children}
    </ThemeProvider>
);

export default MerchantTheme;
export {
    PropsType,
};
