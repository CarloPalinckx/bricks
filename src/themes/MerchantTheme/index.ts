export { default } from './MerchantTheme.template';
