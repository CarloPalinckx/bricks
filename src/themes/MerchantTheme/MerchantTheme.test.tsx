import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import MerchantTheme from '../MerchantTheme';
import { PropsType } from './MerchantTheme.template';

describe('MerchantTheme', () => {
    it('should render the merchant theme', () => {
        const HeadingHierarchyThemeType = {
            fontFamily: 'mockValue',
            fontSize: 'mockValue',
            fontWeight: {
                light: 'mockValue',
                default: 'mockValue',
            },
            lineHeight: 'mockValue',
            color: 'mockValue',
        };

        const theme = {
            Heading: {
                hierarchy1: HeadingHierarchyThemeType,
                hierarchy2: HeadingHierarchyThemeType,
                hierarchy3: HeadingHierarchyThemeType,
                hierarchy4: HeadingHierarchyThemeType,
                hierarchy5: HeadingHierarchyThemeType,
                hierarchy6: HeadingHierarchyThemeType,
            },
        };

        const component = shallow(<MerchantTheme theme={theme as PropsType['theme']} />);

        expect(toJson(component)).toMatchSnapshot();
    });
});
