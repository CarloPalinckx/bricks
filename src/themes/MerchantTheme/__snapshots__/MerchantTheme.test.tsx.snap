// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MerchantTheme should render the merchant theme 1`] = `
<ThemeProvider
  theme={
    Object {
      "Heading": Object {
        "hierarchy1": Object {
          "color": "mockValue",
          "fontFamily": "mockValue",
          "fontSize": "mockValue",
          "fontWeight": Object {
            "default": "mockValue",
            "light": "mockValue",
          },
          "lineHeight": "mockValue",
        },
        "hierarchy2": Object {
          "color": "mockValue",
          "fontFamily": "mockValue",
          "fontSize": "mockValue",
          "fontWeight": Object {
            "default": "mockValue",
            "light": "mockValue",
          },
          "lineHeight": "mockValue",
        },
        "hierarchy3": Object {
          "color": "mockValue",
          "fontFamily": "mockValue",
          "fontSize": "mockValue",
          "fontWeight": Object {
            "default": "mockValue",
            "light": "mockValue",
          },
          "lineHeight": "mockValue",
        },
        "hierarchy4": Object {
          "color": "mockValue",
          "fontFamily": "mockValue",
          "fontSize": "mockValue",
          "fontWeight": Object {
            "default": "mockValue",
            "light": "mockValue",
          },
          "lineHeight": "mockValue",
        },
        "hierarchy5": Object {
          "color": "mockValue",
          "fontFamily": "mockValue",
          "fontSize": "mockValue",
          "fontWeight": Object {
            "default": "mockValue",
            "light": "mockValue",
          },
          "lineHeight": "mockValue",
        },
        "hierarchy6": Object {
          "color": "mockValue",
          "fontFamily": "mockValue",
          "fontSize": "mockValue",
          "fontWeight": Object {
            "default": "mockValue",
            "light": "mockValue",
          },
          "lineHeight": "mockValue",
        },
      },
    }
  }
/>
`;
